### ISSUE TITLE
Parsing fails on psql variable placeholders in COPY commands

### ISSUE THREAD

**Issue by Sanket17052006** [sqlfluff/sqlfluff#7696]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

While I was parsing a PostgreSQL script that uses `psql` variable placeholders (like `:source`), `sqlfluff` throws a `PRS` error. While version 4.1.0 added great support for `\copy` and `\set` meta-commands, it still fails when it encounters a colon-prefixed variable in a standard `COPY` statement.

### Expected Behaviour

It should either ignore or successfully parse `psql` variable placeholders in valid PostgreSQL commands without throwing an unparsable section error.

### Observed Behaviour

`sqlfluff` throws a `PRS` (Found unparsable section) error at the exact location of the `:source` variable. 

```

$ sqlfluff lint --dialect postgres test_source.sql
== [test_source.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'COPY
                       | tmp_table FROM :source\nWITH delimit...'
```

### How to reproduce

1. Create a file named `test_source.sql` with the following content:
   ```sql
   COPY tmp_table FROM :source
   WITH delimiter E'\t' quote E'\b' csv;
   ```
2. Run the following command in your terminal:
   ```bash
   sqlfluff lint --dialect postgres test_source.sql
   ```

### Dialect

postgres

### Version

4.1.0

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
